zamba/models/config.py

```diff
@@ -495,19 +496,28 @@ def preprocess_labels(cls, values):
                     labels["site"], proportions=values["split_proportions"]
                 )
             else:
-                logger.info(
-                    "No 'site' column found so videos will be randomly allocated to splits."
-                )
                 # otherwise randomly allocate
-                random.seed(SPLIT_SEED)
-                labels["split"] = random.choices(
-                    list(values["split_proportions"].keys()),
-                    weights=list(values["split_proportions"].values()),
-                    k=len(labels),
+                logger.info(
+                    "No 'site' column found so videos will be randomly allocated using split proportions."
                 )
 
+                expected_labels = [k for k, v in values["split_proportions"].items() if v > 0]
+                labels["split"] = ""
+                seed = SPLIT_SEED
+
+                while len(np.setdiff1d(expected_labels, labels.split.unique())):
```

Comment:
This has the (improbable) possibility of running forever. Does it make more sense to group by the labels and then assign each video in that group randomly to a split, warning when the group is less than three items?

Reply:
Just for clarity, I do mean the _labels_ that the videos have here (not "train", "val" and "holdout")

Reply:
this can get tricky too. say i have 5 examples of 20 different species (100 videos) and i set my split proportions to be 8/1/1. if i group by species and assign to sets based on 8/1/1 probabilities, all will go to train. however, if i do this across the whole labels file (not within species), these proportions will be valid.

Reply:
curious what your desired behavior would be given that data setup

Reply:
you could heuristically do 1 video in each to start and then random after that

Reply:
clever